addons/mail/models/mail_message.py

```diff
@@ -504,15 +476,19 @@ def message_format(self):
 
         # fetch notification status
         notif_dict = {}
```

Comment:
using a defaultdict will improve code readability

Reply:
@Xavier-Do 
https://github.com/odoo/odoo/pull/30067/commits/d2b247c39b5fd518183f95a8736a9b59e701a545#diff-86c5f1ed15bfa23d6be2285dc88e8a57R480
is it ok for you?
As we have nested dict, so no need to check if `history_partner_ids` or `needaction_partner_ids` in `notif_dict[mid]` and it will take its default value.
Thanks

Reply:
Yes that was the idea
you can also replace

`message['needaction_partner_ids'] = notif_dict.get(message['id'], dict()).get('needaction_partner_ids', [])`

by

`message['needaction_partner_ids'] = notif_dict[message['id']]['needaction_partner_ids']`

Maybe an message.update(notif_dict[message['id']]) could also work for both fields, but it would be less readable and I didn't test it.

Reply:
@Xavier-Do 
Changes Done.
https://github.com/odoo/odoo/pull/30067/commits/21c97e052ba76f31edac48b07c7a730f1d6ec2b3 and I think this one is more readable than the previous one.

Reply:
LGTM, I'll make a complete check tbefore the end of the week to be sure but looks like it' ok like that: good job! 

I just took a quick look to ('res_partner_id.partner_share', '=', False) in the cron, it is correct, but expensive. 

I think that the previous partner.share check in the mark as read methods were not useful: as far as I can see, it was some dead code from previous versions. We made some quick tests on our prod databases, and there are no remaining notification matching is_read=True and is_email=False after January 2018 (looking at message create date). I have to check that in detail, you can do it too, but if a user is partner.share, notification will always be of type email. Thus this check seems not useful and could maybe just replace it by a comment.

Reply:
@Xavier-Do 
Done.
yes, I have already checked several scenario to delete notifications through the cron job, and it's working as expected (older notifications with partner and customer)
Thanks